src/kit.ts

```diff
@@ -739,6 +739,26 @@ export function descriptionForKit(kit: Kit) {
   return 'Unspecified (Let CMake guess what compilers and environment to use)';
 }
 
+/**
+ * Process parameters of the kit and perform any desired transformations.
+ *
+ * Right now, this resolves environment variables used in the tool chain file. This logic could be expanded elsewhere to enable environment variable resolution elsewhere.
+ *
+ * May no longer be necessary if we can leverage some part of https://github.com/Microsoft/vscode/issues/2809 should it be completed and shipped.
+ *
+ * @param kit A kit json structure
+ */
+export function processKit(kit: Kit) {
+  if (kit.toolchainFile) {
+    // map ${foo} -> process.env[foo] to resolve corresponding environment variables
+    kit.toolchainFile = kit.toolchainFile.replace(/\$\{(\w+)\}/, (_str: string, p1: string, ..._args) => {
```

Comment:
Have you checked whether case sensitivity is an issue here?
This here might be useful: https://github.com/vector-of-bool/vscode-cmake-tools/blob/develop/src/expand.ts